src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -664,6 +671,109 @@ def get_no_wait(self) -> bool:
                 no_wait = False
         return no_wait
 
+    # pylint: disable=unused-argument
+    def _get_outbound_type(
+        self,
+        enable_validation: bool = False,
+        read_only: bool = False,
+        load_balancer_profile: ManagedClusterLoadBalancerProfile = None,
+        **kwargs
+    ) -> Union[str, None]:
+        """Internal function to dynamically obtain the value of outbound_type according to the context.
+
+        Note: Inherited and extended in aks-preview to add support for the newly added nat related constants.
```

Comment:
As depicted here, this getter is overwritten in aks-preview to add support for the newly added nat related constants.